libvast/src/type.cpp

```diff
@@ -789,6 +789,83 @@ data type::construct() const noexcept {
   return *this ? caf::visit(f, *this) : data{};
 }
 
+data type::to_definition() const noexcept {
+  // Utility function for adding the attributes to a type definition, if
+  // required.
+  auto attributes_enriched_definition
+    = [&](data type_definition) noexcept -> data {
+    auto attributes_definition = list{};
+    for (const auto& [key, value] : attributes(recurse::no)) {
+      if (value.empty())
+        attributes_definition.push_back(std::string{key});
+      else
+        attributes_definition.push_back(
+          record{{std::string{key}, std::string{value}}});
+    }
+    if (attributes_definition.empty())
+      return type_definition;
+    return record{
+      {"type", std::move(type_definition)},
+      {"attributes", std::move(attributes_definition)},
+    };
+  };
+  // Check if there is an alias, and if there is then visit then one first.
+  for (const auto& alias : aliases()) {
+    auto definition = attributes_enriched_definition(alias.to_definition());
+    const auto name = this->name();
+    if (name.empty())
+      return definition;
+    return record{
+      {std::string{name}, std::move(definition)},
+    };
+  }
```

Comment:
Doing this inline means that all use sites of aliases will be replaced by their definition. Wouldn't it be a bit nicer to do this only for unnamed aliases and collecting the named ones in a set to print there definitions separately?

Reply:
That could be a future extension, but I'd much rather like to see that as an operation on multiple type definitions to form a whole module.